src/dbus/server/dbus_thread_object_rcp.cpp

```diff
@@ -1957,6 +1960,19 @@ otError DBusThreadObjectRcp::SetNat64Cidr(DBusMessageIter &aIter)
 }
 #endif // OTBR_ENABLE_NAT64
 
+void DBusThreadObjectRcp::SetAdminPasscodeEnabled(DBusRequest &aRequest)
+{
+    otError error = OT_ERROR_NONE;
+    bool    enable;
+    auto    args = std::tie(enable);
+
+    VerifyOrExit(DBusMessageToTuple(*aRequest.GetMessage(), args) == OTBR_ERROR_NONE, error = OT_ERROR_INVALID_ARGS);
+    mBorderAgent.SetEpskcEnabled(enable);
```

Comment:
I'd suggest to do following instead:

- In ot-br-posix's `BorderAgent`, use [otBorderAgentSetEphemeralKeyCallback](https://github.com/openthread/openthread/blob/20c9bcbc4142fd74e490bd50faa7def9e1b3a823/include/openthread/border_agent.h#L275) to listen to the Epskc state changes and updates the meshcop service accordingly.
- In `DBusThreadObjectRcp::SetAdminPasscodeEnabled`, just enable/disable the Epskc mode by `otBorderAgentSetEphemeralKey`.

In this way you have a single source of truth from OT Core and don't have to maintain the state at ot-br-posix's `BorderAgent`. Also you don't have to let `DBusThreadObjectRcp` control `BorderAgent`.

Reply:
This PR is to update the `epskc capability` bit in state bitmap of meshcop service according to feature flags, other than publish/unpublish meshcop-e service according to epskc setting.

Reply:
Thanks for the context! I misinterpreted this as the API to active/deactivate the Epskc mode. 

Since that's for the capability feature flag, I'd suggest to directly add a new field in [feature flag proto instead of adding a new D-BUS method. New D-BUS method introduces more fragmentations of the feature flag framework.

https://github.com/openthread/ot-br-posix/blob/b7ebdc361c4c051448b1c611e9dde3d20c1bc5c7/src/ncp/rcp_host.cpp#L231C18-L231C30